src/Pages/Authentication/signup.js

```diff
@@ -0,0 +1,62 @@
+import { useAuth } from '../../Context/authContext'
+
+const Signup = () => {
+
+    const[userDetails, setUserDetails] = useState({firstName: '', lastName: '', email: '', password: '', confirmPassword: '', terms: false})
+    const [error, setError] = useState({isError: false, text: ''})
+    const {authState, signUp} = useAuth();
+
+    const submitHandler = ()=>{
+
+        
+        if(!userDetails.firstName || !userDetails.lastName || !userDetails.email || !userDetails.password || !userDetails.confirmPassword || !userDetails.terms){
+            setError({isError: true, text: 'Please Enter all the fields'})
+        }else if(!userDetails.email.includes('@')){
+            setError({isError: true, text: 'Please Enter valid email id'})
+        }else if(userDetails.password !== userDetails.confirmPassword){
+            setError({isError: true, text: 'Password and confirm password do not match'})
+        }else{
+            signUp(userDetails)
+            setUserDetails({firstName: '', lastName: '', email: '', password: '', confirmPassword: '', terms: false})
+        }
+    }
+
+  return (
+    <div className="wrapper" style={{marginLeft: '30rem'}}>
+        <h2 style={{fontSize: '2rem', marginTop: '-2.5rem'}}>Sign-Up</h2>
+        {error.isError && <span style={{color: 'red'}}>{error.text} !!</span>}
+        <form onSubmit={(e)=>{e.preventDefault()}}>
+            <div className="input-box">
+                <input type="text" placeholder="Enter your first name" value={userDetails.firstName} onChange={(e)=>setUserDetails({...userDetails, firstName: e.target.value})}  />
+            </div>
+            <div className="input-box">
+                <input type="text" placeholder="Enter your last name" value={userDetails.lastName} onChange={(e)=>setUserDetails({...userDetails, lastName: e.target.value})} />
+            </div>
+            <div className="input-box">
+                <input type="text" placeholder="Enter your email" value={userDetails.email} onChange={(e)=>setUserDetails({...userDetails, email: e.target.value})} />
+            </div>
+            <div className="input-box">
+                <input type="password" placeholder="Create password" value={userDetails.password} onChange={(e)=>setUserDetails({...userDetails, password: e.target.value})} />
+            </div>
+            <div className="input-box">
+                <input type="password" placeholder="Confirm password" value={userDetails.confirmPassword} onChange={(e)=>setUserDetails({...userDetails, confirmPassword: e.target.value})} />
+            </div>
+            <div className="policy">
+                <input type="checkbox" value={userDetails.terms} onChange={(e)=>setUserDetails({...userDetails, terms: e.target.value})} />
+                <h3>I accept all terms & condition</h3>
+            </div>
+            <div className="input-box" >
+                <input type="Submit" value="Register Now" onClick={()=>submitHandler()}/>
```

Comment:
suggestion - if there are no arguments to `submitHandler` then you can simply write `onClick={submitHandler}` and that would work too.